LiskHQ/lisk
# Improve chain connector storage

### What was the problem?

This PR resolves #9195 

### How was it solved?

- [x] Improve storage to retrieve by key/val and delete efficiently instead of managing full array
- [x] Reduce number of RPC calls during new block event
- [x] Improve cleanup

### How was it tested?

- Run [interop setup](https://github.com/LiskHQ/lisk-sdk/tree/development/examples/interop)
- Register sidechains on mainchain and mainchain on sidechains.
- Make cross chain transfers using the scripts [as mentioned in the readme](https://github.com/LiskHQ/lisk-sdk/blob/development/examples/interop/README.md)
- Stop mainchain-node-two for more than 1000 blocks (mainchain-node-one generates the blocks) and then start it and see if it syncs properly and starts sending valid CCUs without any errors.
